assertj-core/src/main/java/org/assertj/core/api/AbstractStringAssert.java

```diff
@@ -402,4 +408,279 @@ public SELF isEqualTo(String expectedStringTemplate, Object... args) {
   public SELF isEqualTo(String expected) {
     return super.isEqualTo(expected);
   }
+
+  /**
+   * Parses the actual value as boolean, the parsed boolean becoming the new value under test.
+   * <p>
+   * Note that only when the string is equal to the string "true", ignoring case, and can have leading and trailing space, the parsed value will be {@code true}.
+   * Other not null strings will make the value be {@code false}.
+   * <p>
+   * Examples:
+   * <pre><code class='java'> // assertion succeeds
+   * assertThat(&quot;truE&quot;).asBoolean().isTrue();
+   * assertThat(&quot;false&quot;).asBoolean().isFalse();
+   * assertThat(&quot;foo bar&quot;).asBoolean().isFalse();
+   *
+   * // assertion fails as the actual value is null
+   * assertThat((String) null).asBoolean();</code></pre>
+   *
+   * @return a new {@link BooleanAssert} instance whose value under test is the result of the parse.
+   * @throws AssertionError if the actual value is {@code null}.
+   *
+   * @since 3.24.0
```

Comment:
just because we were slow discussing the issue and  looking at this PR, this is now for 3.25.0 :sweat_smile: